sympy/plotting/plot.py

```diff
@@ -2135,7 +2138,36 @@ def check_arguments(args, expr_len, nb_of_free_symbols):
             i = len(args) + 1
 
         exprs = Tuple(*args[:i])
-        free_symbols = list(set().union(*[e.free_symbols for e in exprs]))
+        # get all variables
+        variables = set().union(*[e.free_symbols for e in exprs])
+        free_symbols = list(variables)
+        # check arguments
+        assert all([isinstance(rang, Tuple) for rang in args[expr_len:]]), \
+            "Ranges should be in the form (symbol, a, b)"
+        # get symbols from all ranges
+        specify_symbols = set([s[0] for s in args[expr_len:]])
+
+        style_ranges = ",  {} = range(*)"*len(specify_symbols)
+        expr_ranges = style_ranges.format(*specify_symbols)[3:]
+
+        all_symbols = variables | specify_symbols
+        if len(all_symbols) > len(variables):
+            msg = "There be a variable ranged that isn´t part of any ecuation!"
+            raise NameError(msg)
+
+        if len(all_symbols) > nb_of_free_symbols:
+            if specify_symbols == set():
+                msg =  "There must be  ONLY %s (%s) parameters from %s"%(
+                    get_text(nb_of_free_symbols),
+                    nb_of_free_symbols, all_symbols)
+                raise NameError(msg)
+
+            msg = "There must be  ONLY %s (%s) parameters, "       \
+                "but was given %s.  %s!"%(
```

Comment:
This message might also look better formatted like the previous suggestion.

Also, what does the `expr_ranges` depict after the warning message?
Is displaying the ranges after the message helpful in some way?

Reply:
When you have more than one line in the message, `utilities.misc.filldedent` can help:
```
msg = filldedent("""
    This is the message and it is
    going to get parameter %s and
    say a little more""" % p)
```